packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,176 @@
+import re
+import subprocess
+from datetime import datetime
+from typing import Dict, List
+import os
+
+from github import Repository, PullRequest, Commit
+from jinja2 import Environment, FileSystemLoader
+
+from .common_validations import (is_tag, is_version)
+from git import Repo
+import pathlib2
+from typing import Tuple
+
+BASE_GIT_PATH = pathlib2.Path(__file__).parents[1]
+PATCH_VERSION_MATCH_FROM_MINOR_SUFFIX = "\.\d{1,3}"
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_project_version_from_tag_name(tag_name: is_tag(str)) -> str:
+    return tag_name[1:]
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def find_nth_occurrence_position(subject_string: str, search_string: str, n) -> int:
+    start = subject_string.find(search_string)
+
+    while start >= 0 and n > 1:
+        start = subject_string.find(search_string, start + 1)
+        n -= 1
+    return start
+
+
+def find_nth_matching_line_and_line_number(subject_string: str, regex_pattern: str, n: int) -> Tuple[int, str]:
+    """Takes a subject string, regex param and the search index as parameter and returns line number of found match.
+    If not found returns -1"""
+    lines = subject_string.splitlines()
+    counter = 0
+    for line_number, line in enumerate(lines):
+        if re.match(regex_pattern, line):
+            counter = counter + 1
+        if counter == n:
+            return line_number, lines[line_number]
+    return -1, ""
+
+
+def remove_text_with_parenthesis(param: str) -> str:
+    """Removes texts within parentheses i.e. outside parenthesis(inside parenthesis)-> outside parenthesis """
```

Comment:
two minor comments:

* Removes texts within parenthesis **together with the parenthesis itself** ..
* parentheses -> parenthesis